### PR TITLE
fix: allow DT warehouse to be updated in-place

### PR DESCRIPTION
**Pull request by sonya**
Dynamic table warehouses can be updated in-place with an ALTER DYNAMIC TABLE statement. The terraform resource was requiring the DT to be dropped and recreated in order to change the warehouse, but this is not necessary.

This commit removes the ForceNew modifier on the "warehouse" parameter so that users can easily change the warehouse. This is a setting that users may want to easily change for cost allocation purposes.

## Test Plan
* [ ] tested by running `terraform apply` and `terraform plan` on an existing stack with warehouse changes, observed that dynamic tables were updated in place, and `ALTER DYNAMIC TABLE` was run in Snowflake

## References

Documentation for `ALTER DYNAMIC TABLE` command: https://docs.snowflake.com/en/sql-reference/sql/alter-dynamic-table

### CODE REVIEW

**Comment by sfc-gh-asawicki** [comment]
Hey @sonya. Thanks for the PR. This week is pretty tough for us, I will check the PR next week.

**Review by sfc-gh-jcieslak — DISMISSED** [review]

File: pkg/resources/dynamic_table.go [hunk 1]
```diff
@@ -61,7 +61,6 @@ var dynamicTableShema = map[string]*schema.Schema{
 		Type:        schema.TypeString,
 		Required:    true,
 		Description: "The warehouse in which to create the dynamic table.",
-		ForceNew:    true,
```
Comment (sfc-gh-asawicki) [inline pkg/resources/dynamic_table.go:64]: Please add the `warehouse` param change in the `TestAcc_DynamicTable_basic` test.

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]
Hey @sonya. Thanks for the contribution.

I have just a small comment with the missing test, and we can merge it after!

**Comment by sonya** [comment]
Added the field to the basic acceptance test. I'm not sure if that's sufficient to demonstrate that the DT doesn't get destroyed and recreated. I did some more work in #2437 on being able to tell if a change caused a replace (but I might be missing a much more obvious way)

**Comment by sfc-gh-asawicki** [comment]
Thanks, I will check it today.

**Review by sfc-gh-jcieslak — APPROVED** [review]
Maybe the second warehouse could be named TestSecondaryWarehouseName, but It's just a cosmetic change and we can do it later. Thanks for the contribution!

**Review by sfc-gh-asawicki — APPROVED** [review]

**Comment by sfc-gh-asawicki** [comment]
I have merged the change. Thank you for your contribution @sonya!

**Pull request merged**